dangerfile.ts

```diff
@@ -37,3 +40,53 @@ if (!hasChangelog) {
     }
   })
 }
+
+/* === Package dependencies ========================================================================================= */
+
+let dependenciesChangedLogged = false
+
+function logDependenciesChanged() {
+  if (!dependenciesChangedLogged) {
+    dependenciesChangedLogged = true
+    warn('Package dependencies changed. Make sure you have OSS approval before merging!')
+  }
+}
+
+danger.git.created_files
+  .filter(filepath => filepath.match(/\bpackage\.json$/))
+  .forEach(filepath => {
+    warn(`New package added: ${filepath}. Make sure you have OSS approval before merging!`)
+  })
+
+danger.git.modified_files
+  .filter(filepath => filepath.match(/\bpackage\.json$/))
+  .forEach(async filepath => {
+    const diff = await danger.git.JSONDiffForFile(filepath)
+    if (diff.dependencies) {
+      const before = { ...diff.dependencies.before, ..._.zipObject(diff.dependencies.added) }
+      const after = diff.dependencies.after || {}
+      const changedDependencies = _.reduce(
+        before,
+        (result, value, key) => {
+          return value === after[key] || !after[key]
+            ? result
+            : { ...result, [key]: { before: value, after: after[key] } }
+        },
+        {},
+      )
```

Comment:
I think that we can extract this to a separate function, i.e. `getChangedDependencies(filepath)`